dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/FileFormatHelper.java

```diff
@@ -66,4 +75,20 @@ public static void updateFileFormats(Set<? extends Version> versions, final File
         });
         return fileFormatsFromDB;
     }
+
+    public static Optional<String> calcSHA1(String content) {
+        if (content != null) {
+            try {
+                MessageDigest messageDigest = MessageDigest.getInstance("SHA-1");
+                messageDigest.update(content.getBytes(StandardCharsets.UTF_8), 0, content.length());
```

Comment:
Missed this the first time; `content.length` is wrong; you need the length of the byte array, something like:

```
final byte [] bytes = content.getBytes(StandardCharsets.UTF_8);
messageDigest.update(bytes, 0, bytes.length);
```

Explanation: `content.length()` returns the number of characters; a character can be made of more than one byte, so the bytes array length might be longer the number of characters. Typically not an issue for "standard" English characters, but throw in some accented characters or non-Latin characters, and it will become one.